aspnetcore/host-and-deploy/health-checks.md

```diff
@@ -5,13 +5,728 @@ description: Learn how to set up health checks for ASP.NET Core infrastructure,
+{
+    endpoints.MapHealthChecks("/health");
+});
+```
+
+Health Checks Middleware is a *terminal middleware* in the app's request processing pipeline. The first health check endpoint encountered that's an exact match to the request URL executes and short-circuits the rest of the middleware pipeline. When short-circuiting occurs, no middleware following the matched health check executes.
+
+### Require host
+
+Call `RequireHost` to specify one or more permitted hosts for the health check endpoint. Hosts should be Unicode rather than punycode and may include a port. If a collection isn't supplied, any host is accepted.
+
+```csharp
+app.UseEndpoints(endpoints =>
+{
+    endpoints.MapHealthChecks("/health").RequireHost("www.contoso.com:5001");
+});
+```
+
+For more information, see the [Filter by port](#filter-by-port) section.
+
+### Require authorization
+
+Call `RequireAuthorization` to run Authorization Middleware on the health check request endpoint. A `RequireAuthorization` overload accepts one or more authorization policies. If a policy isn't provided, the default authorization policy is used.
+
+```csharp
+app.UseEndpoints(endpoints =>
+{
+    endpoints.MapHealthChecks("/health").RequireAuthorization();
+});
+```
+
+### Enable Cross-Origin Requests (CORS)
+
+Call `RequireCors` to run CORS Middleware on the health check request endpoint. A `RequireCors` overload accepts a CORS policy builder delegate (`CorsPolicyBuilder`) or a policy name. If a policy isn't provided, the default CORS policy is used.
+
+In `Startup.ConfigureServices`:
+
+```csharp
+services.AddCors(options =>
+{
+    options.AddPolicy("MyAllowSpecificOrigins",
+        builder =>
+        {
+            builder.WithOrigins("http://example.com",
+                                "http://www.contoso.com");
+        });
+});
+```
+
+In `Startup.Configure`:
+
+```csharp
+app.UseEndpoints(endpoints =>
+{
+    endpoints.MapHealthChecks("/health").RequireCors("MyAllowSpecificOrigins");
+});
+```
+
+For more information, see <xref:security/cors>.
+
+## Health check options
+
+<xref:Microsoft.AspNetCore.Diagnostics.HealthChecks.HealthCheckOptions> provide an opportunity to customize health check behavior:
+
+* [Filter health checks](#filter-health-checks)
+* [Customize the HTTP status code](#customize-the-http-status-code)
+* [Suppress cache headers](#suppress-cache-headers)
+* [Customize output](#customize-output)
+
+### Filter health checks
+
+By default, Health Check Middleware runs all registered health checks. To run a subset of health checks, provide a function that returns a boolean to the <xref:Microsoft.AspNetCore.Diagnostics.HealthChecks.HealthCheckOptions.Predicate> option. In the following example, the `Bar` health check is filtered out by its tag (`bar_tag`) in the function's conditional statement, where `true` is only returned if the health check's <xref:Microsoft.Extensions.Diagnostics.HealthChecks.HealthCheckRegistration.Tags> property matches `foo_tag` or `baz_tag`:
```

Comment:
middleware **_versus_** routing

Can you explain that out a bit more? To this point, this *only* made sense to me thinking about it as one thing: middleware. I don't follow on the *routing* part because as u said: routing *doesn't run health checks*.

Reply:
> middleware versus routing

I think showing people this code:

```C#
app.UseEndpoints(endpoints =>
{
    endpoints.MapHealthChecks(...);
});
```

and saying "Health checks middleware" is confusing.

**technically** the health checks middleware is being used - because you're using routing to route to the health checks middleware. So I guess if we want to say "using routing to run the health checks middleware" - I'd be good with that.

Reply:
I took a look ...

Most (almost all) of the `MapHealthChecks` content (text) speaks to 'creating an endpoint'. The sixteen references to the middleware speak to behaviors that the middleware exhibits or to facts about the middleware (e.g., configuration).

I'm done for the next commit. Sending it up now .............